app/main.py

```diff
@@ -2,0 +1,54 @@
+class Animal:
+    alive = []
+
+    def __init__(self, name, health: int = 100, is_hidden: bool = False):
+        self.name = name
+        self.health = health
+        self.is_hidden = is_hidden
+        self.alive.append(self)
+
+
+    def die(self):
+        if self in Animal.alive:
+            self.alive.remove(self)
```

Comment:
Similar to the constructor, `alive` should be accessed as a class variable. Use `Animal.alive.remove(self)` instead.